realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -261,10 +261,13 @@ public boolean isEmpty() {
 
     /**
      * Clears the table i.e., deleting all rows in the table.
+     *
+     * WARNING: This method is not supported by Partial Sync, use `findAll().deleteFromRealm`
```

Comment:
It kind of is supported now, no? With magic 😄

Reply:
The server isn't enforcing it right now. I believe it just ignores the instruction.

Reply:
No I'm asking because you seem to have just added the means to clear via `isPartial == true` in JNI side

Reply:
There is a subtle difference `table->where().find_all().clear()` creates a query result and delete the individual objects in it. Resulting in a lot of `delete_row` instructions. Which is also why it is slower.

`table->clear()` just results in one `clearTable` instruction, which isn't supported by partial sync right now (mostly because "the whole table" is a loose concept in a world with partial sync.

Reply:
I think  @Zhuinden point was that the comment doesn't makes sense since you implemented internally a fallback to `table->where().find_all().clear()`

I think it should rephrase it to something like 
> WARNING: In the context of partial sync, this method will behave similarly to a `findAll().deleteFromRealm()` which might have a performance impact.